freshmaker/pyxis.py

```diff
@@ -215,6 +215,135 @@ def get_images_by_digests(self, digests):
         :return: bundle ContainerImages
         :rtype: list
         """
-        request_params = {'include': 'data.brew',
-                          'filter': f"repositories.manifest_list_digest=in=({','.join(digests)})"}
-        return self._pagination("images", request_params)
+        request_params = {'include': 'data.brew,data.repositories',
+                          'filter': f'repositories.manifest_list_digest=in=({",".join(digests)})'}
+        return self._pagination('images', request_params)
+
+    def _add_repositories_info(self, reg_repo_info):
+        """
+        For every pair of registry-repository add information about it's
+        auto_rebuild tags. To decrease amount of queries to Pyxis, only one
+        query is performed with the filter set to proper registry-repository
+        pair.
+
+        Tags list is added to the parameter object of the method.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        """
+        if not reg_repo_info:
+            return None
+        fltr = ""
+        # Construct filter for future request to Pyxis with registry-repository pairs
+        for reg, repo in reg_repo_info.keys():
+            if not fltr:
+                fltr = f"(registry=={reg};repository=={repo})"
+            else:
+                fltr += f",(registry=={reg};repository=={repo})"
+        params = {'include': ','.join(['data.auto_rebuild_tags',
+                                       'data.registry', 'data.repository']),
+                  'filter': fltr}
+        repos = self._pagination('repositories', params)
+
+        # For every repo add it's auto_rebuild_tags info
+        for repo in repos:
+            reg_repo_pair = (repo['registry'], repo['repository'])
+            # one of repos isn't in previously constructed map,
+            # so there is inconsistency
+            if reg_repo_pair not in reg_repo_info:
+                log.warning("There is inconsistency in naming for: %s/%s",
+                            reg_repo_pair[0], reg_repo_pair[1])
+                continue
+            tags = repo.get('auto_rebuild_tags')
+            # If the repository doesn't have 'auto_rebuild_tags', don't proceed with it
+            if tags:
+                reg_repo_info[reg_repo_pair]['tags'] = set(tags)
```

Comment:
Could you please use the key `auto_rebuild_tags` to make this clearer when processing this data outside of this method?